contracts/SignerBondsUniswapV2.sol

```diff
@@ -91,9 +95,21 @@ contract SignerBondsUniswapV2 is ISignerBondsSwapStrategy, Ownable {
         );
     }
 
-    /// @notice Swaps signer bonds.
-    /// @dev Adds incoming bonds to the overall contract balance.
-    function swapSignerBonds() external payable override {}
+    /// @notice Receive ETH upon withdrawal of risk manager's signer bonds.
+    /// @dev Do not send arbitrary funds. They will be locked forever.
+    receive() external payable {}
+
+    /// @notice Swaps the given signer bonds amount from the given risk manager.
+    /// @dev Swaps incoming signer bonds on Uniswap V2.
+    /// @param riskManager Address of the risk manager
+    /// @param amount Amount of signer bonds being swapped.
+    function swapSignerBonds(RiskManagerV1 riskManager, uint256 amount)
```

Comment:
Given that this function accepts now risk manager as a parameter so it is not bound to a single implementation of a risk manager and we can use the same `SignerBondsUniswapV2` contract for, for example, both tBTC v1 and tBTC v2 risk managers, I would rather define an interface (`IWithdrawable`?) in `SignerBondsUniswapV2` with `withdrawSignerBonds` and I'd accept this interface here instead of `RiskManagerV1`.

Reply:
There are several problems with the approach you propose. First, we don't know if `signer bonds` concept will be present in tBTC v2. Secondly, `IWithdrawable` is not correct as we also need to check `hasOpenAuctions` on the risk manager. We can extract an `IRiskManager` interface but having `hasOpenAuctions` here feels bad as auctions are not a risk manager concept. Also, we must have that interface in `interfaces` dir to avoid cyclic dependencies as both strategies import `ISignerBondsSwapStrategy` from `RiskManagerV1.sol`. Another option is moving the open auctions check into the risk manager itself. We could, for example, revert `withdrawSignerBonds` if there are open auctions. However, I still have a feeling we're mixing unrelated concepts here. Have you any other ideas how we can solve the problem here?

Reply:
>  We can extract an `IRiskManager` interface but having `hasOpenAuctions` here feels bad as auctions are not a risk manager concept.

I... think it is fine. The risk manager is opening auctions and is managing their length. It also has `onAuctionFullyFilled` and `onAuctionPartiallyFilled` so although it is not the component managing the auctions, we can say that it has open auctions - the auctions it opened to gather funds for purchasing signer bonds...

I'd vote for `IRiskManager` with `hasOpenAuctions` personally.

Another option is to accept that the swap strategy does not work with any risk manager but with `RiskManagerV1` explicitly, and make `RiskManagerV1` a constructor parameter of the strategy. This creates a chicken and egg problem though so we would need to remove swap strategy from the risk manager's constructor parameters and let to initiate the risk manager with the first swap strategy with no delay (same as we do for risk manager in coverage pool) but... this feels too complicated.

Reply:
Ok here it is https://github.com/keep-network/coverage-pools/pull/88/commits/5c74fa0af23afe774a52c87a3769e73fd07f2976